### ISSUE TITLE
assert Undefined with Cypress >= 7.0

### ISSUE THREAD

**Issue by Cedric-Brunel** [cypress-io/cypress#16303]
I just upgraded to version 7.0 and since then I have an assertion that does not work anymore

Extrait Code : 
```javascript
cy.contains("tr",elementRowContain)
   .within(() => {
    cy.get(icon).click({ force: true });
});
```

Résult : 
![image](https://user-images.githubusercontent.com/62300093/116860057-422b2280-ac01-11eb-94df-d927868799b0.png)

J'ai testé ça , toutes les assertions possible :( : 
```javascript
  cy.contains("tr", elementRowContain).
    should("not.be.undefined")
    .and("exist")
    .and("be.visible")
....

```

Résultat  
![image](https://user-images.githubusercontent.com/62300093/116858695-15760b80-abff-11eb-9712-7c9c5a5cd0d0.png)

Why does he put that it is not visible, when I put the should("be.visible") to wait until it is visible? 

My code cannot be executed as is. 
But in summary I get a line from a grid that contains such and such a value.
and then I click on an icon of this line.

Thank's for your help !!!

Sorry for my bad english :(

**Comment by Cedric-Brunel** [comment]
I found the beginning of an explanation! This code is executed in the After hook ==> HS

If I do it elsewhere in a usecase ==> OK

**Comment by jennifer-shehane** [comment]
@Cedric-Brunel Which version of Cypress were you previously on?

**Comment by Cedric-Brunel** [comment]
6.0.1

**Comment by jennifer-shehane** [comment]
@Cedric-Brunel Can you provide a reproducible example? We have the test code, we need the HTML/app/JS, etc to run the code against in order to see the error and see why this happens. It's likely dependent on the application code structure. Thanks.

**Comment by jennifer-shehane** [comment]
I can see how the assertions are not all properly shown, similar to https://github.com/cypress-io/cypress/pull/16218 @sainthkh But need the code to get the undefined error.

<img width="422" alt="Screen Shot 2021-05-03 at 11 53 34 AM" src="https://user-images.githubusercontent.com/1271364/116906692-35f59400-ac06-11eb-8554-99728221a426.png">

```html
<body>
  <div>Foo</div>
</body>
```

```js
it('undefined', () => {
  cy.visit('index.html')
  cy.contains("Foo")
    .should("not.be.undefined")
    .and("contain", "Foo")
    .and("be.visible")
});
```

**Comment by sainthkh** [comment]
It's fixed with #16218. It's not released yet. Maybe this will be fixed with the next version (7.3.0?)

**Comment by Cedric-Brunel** [comment]
ok thank's, but i think, i've another bug.
I try to make an example and i will comeback to you.
